Userland/Libraries/LibCore/Process.cpp

```diff
@@ -160,9 +162,37 @@ ErrorOr<bool> Process::is_being_debugged()
         return (tracer_pid != 0UL);
     }
     return false;
+#elif defined(AK_OS_MACOS)
+    // https://developer.apple.com/library/archive/qa/qa1361/_index.html
+    int junk;
+    int mib[4];
+    struct kinfo_proc info;
+    size_t size;
+
+    // Initialize the flags so that, if sysctl fails for some bizarre
+    // reason, we get a predictable result.
+
+    info.kp_proc.p_flag = 0;
+
+    // Initialize mib, which tells sysctl the info we want, in this case
+    // we're looking for information about a specific process ID.
+
+    mib[0] = CTL_KERN;
+    mib[1] = KERN_PROC;
+    mib[2] = KERN_PROC_PID;
+    mib[3] = getpid();
+
+    // Call sysctl.
+
+    size = sizeof(info);
+    junk = sysctl(mib, sizeof(mib) / sizeof(*mib), &info, &size, NULL, 0);
+    assert(junk == 0);
```

Comment:
Instead of yoinking the error into a junk variable, we should probably return it as a syscall Error?

```
if (syctl(...) < 0)
    return Error::from_syscall("sysctl"sv, -errno);
```

Reply:
Whoops, I just yoinked the code from the apple documentation example, did not pay attention to "C++"-ize it :grin: